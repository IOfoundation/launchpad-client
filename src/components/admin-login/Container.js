import React from 'react';
import {PropTypes} from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';

import SnackbarUI from '@Shared/SnackBar';

import {containerStyles} from '@Utils/containerStyles';

const Container = props => {
  const {classes, children, direction, modifier} = props;

  return (
    <div
      className={['admin-login', classes.container, classes[modifier]].join(
        ' '
      )}
    >
      <SnackbarUI />
      <section className={['content-section', classes.content].join(' ')}>
        <Grid
          container={true}
          justify="center"
          alignItems="center"
          direction={direction}
        >
          {children}
        </Grid>
      </section>
    </div>
  );
};

const styles = theme => ({
  content: {
    ...containerStyles(theme),
    color: 'white',
  },
  container: {
    alignItems: 'center',
    backgroundColor: '#4a4a4a',
    backgroundImage: "url('/static-data/backgrounds/admin-login.png')",
    backgroundPosition: 'center',
    backgroundSize: 'cover',
    display: 'flex',
    height: 'calc(100vh - 48px)',
    justifyContent: 'center',
    [theme.breakpoints.down('sm')]: {
      padding: '2em',
      height: 'auto',
      marginTop: '52px',
    },
  },
  item: {
    marginRight: '2em',
  },
  modifier: {
    alignItems: 'flex-start',
    paddingTop: '131px',
    [theme.breakpoints.down('sm')]: {
      padding: '10em 2em 2em',
    },
  },
});

Container.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  classes: PropTypes.shape({
    content: PropTypes.string,
    container: PropTypes.string,
    item: PropTypes.string,
  }),
  direction: PropTypes.string,
  modifier: PropTypes.string,
};

export default withStyles(styles)(Container);
