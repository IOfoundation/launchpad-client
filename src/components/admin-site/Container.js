import React from 'react';
import {PropTypes} from 'prop-types';

import {withStyles} from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import {containerStyles} from '../../utils/containerStyles';

const styles = theme => ({
  content: {
    ...containerStyles(theme),
    color: 'white',
  },
  container: {
    alignItems: 'center',
    backgroundColor: '#F7F9FD',
    display: 'flex',
    padding: '46px 0',
    justifyContent: 'center',
    [theme.breakpoints.down('sm')]: {
      padding: '2em',
      height: 'auto',
      marginTop: '52px',
    },
  },
});

const Container = props => {
  const {classes, children} = props;

  return (
    <div className={['admin-login', classes.container].join(' ')}>
      <section className={['content-section', classes.content].join(' ')}>
        <Grid container={true} justify="center" alignItems="flex-start">
          {children}
        </Grid>
      </section>
    </div>
  );
};

Container.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  classes: PropTypes.shape({
    container: PropTypes.string,
    content: PropTypes.string,
    item: PropTypes.string,
  }),
};

export default withStyles(styles)(Container);
