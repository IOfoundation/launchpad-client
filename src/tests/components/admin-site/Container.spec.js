import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import Container from '../../../components/admin-site/Container';

describe('Admin-site <Container />', () => {
  it('renders snapshot of Container', () => {
    const wrapper = mount(<Container />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
