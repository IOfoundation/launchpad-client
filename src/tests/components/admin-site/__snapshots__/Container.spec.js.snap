// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin-site <Container /> renders snapshot of Container 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WithStyles(Container)>
    <Container
      classes={
        Object {
          "container": "Container-container-2",
          "content": "Container-content-1",
        }
      }
    >
      <div
        className="admin-login Container-container-2"
      >
        <Connect(WithStyles(SnackbarUI))>
          <WithStyles(SnackbarUI)
            action="DISMISS"
            actions={
              Object {
                "hideSnackbar": [Function],
                "showSnackbar": [Function],
                "testingSnackbar": [Function],
              }
            }
            autoHideDuration={10000}
            message="placeholder message"
            visibility={false}
          >
            <SnackbarUI
              action="DISMISS"
              actions={
                Object {
                  "hideSnackbar": [Function],
                  "showSnackbar": [Function],
                  "testingSnackbar": [Function],
                }
              }
              autoHideDuration={10000}
              classes={
                Object {
                  "close": "SnackbarUI-close-3",
                }
              }
              message="placeholder message"
              visibility={false}
            >
              <WithStyles(Snackbar)
                action={
                  <WithStyles(Button)
                    aria-label="Close"
                    className="snackbar__action"
                    color="inherit"
                    onClick={[Function]}
                    size="small"
                  >
                    DISMISS
                  </WithStyles(Button)>
                }
                autoHideDuration={10000}
                className="snackbar"
                message={
                  <span
                    id="snackbar__message"
                  >
                    placeholder message
                  </span>
                }
                onClose={[Function]}
                open={false}
                style={
                  Object {
                    "bottom": "10px",
                  }
                }
              >
                <Snackbar
                  TransitionComponent={[Function]}
                  action={
                    <WithStyles(Button)
                      aria-label="Close"
                      className="snackbar__action"
                      color="inherit"
                      onClick={[Function]}
                      size="small"
                    >
                      DISMISS
                    </WithStyles(Button)>
                  }
                  anchorOrigin={
                    Object {
                      "horizontal": "center",
                      "vertical": "bottom",
                    }
                  }
                  autoHideDuration={10000}
                  className="snackbar"
                  classes={
                    Object {
                      "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter-6",
                      "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft-10",
                      "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight-8",
                      "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter-5",
                      "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft-9",
                      "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight-7",
                      "root": "MuiSnackbar-root-4",
                    }
                  }
                  disableWindowBlurListener={false}
                  message={
                    <span
                      id="snackbar__message"
                    >
                      placeholder message
                    </span>
                  }
                  onClose={[Function]}
                  open={false}
                  style={
                    Object {
                      "bottom": "10px",
                    }
                  }
                  transitionDuration={
                    Object {
                      "enter": 225,
                      "exit": 195,
                    }
                  }
                />
              </WithStyles(Snackbar)>
            </SnackbarUI>
          </WithStyles(SnackbarUI)>
        </Connect(WithStyles(SnackbarUI))>
        <section
          className="content-section Container-content-1"
        >
          <WithStyles(Grid)
            alignItems="flex-start"
            container={true}
            justify="center"
          >
            <Grid
              alignContent="stretch"
              alignItems="flex-start"
              classes={
                Object {
                  "align-content-xs-center": "MuiGrid-align-content-xs-center-23",
                  "align-content-xs-flex-end": "MuiGrid-align-content-xs-flex-end-25",
                  "align-content-xs-flex-start": "MuiGrid-align-content-xs-flex-start-24",
                  "align-content-xs-space-around": "MuiGrid-align-content-xs-space-around-27",
                  "align-content-xs-space-between": "MuiGrid-align-content-xs-space-between-26",
                  "align-items-xs-baseline": "MuiGrid-align-items-xs-baseline-22",
                  "align-items-xs-center": "MuiGrid-align-items-xs-center-19",
                  "align-items-xs-flex-end": "MuiGrid-align-items-xs-flex-end-21",
                  "align-items-xs-flex-start": "MuiGrid-align-items-xs-flex-start-20",
                  "container": "MuiGrid-container-11",
                  "direction-xs-column": "MuiGrid-direction-xs-column-14",
                  "direction-xs-column-reverse": "MuiGrid-direction-xs-column-reverse-15",
                  "direction-xs-row-reverse": "MuiGrid-direction-xs-row-reverse-16",
                  "grid-lg-1": "MuiGrid-grid-lg-1-82",
                  "grid-lg-10": "MuiGrid-grid-lg-10-91",
                  "grid-lg-11": "MuiGrid-grid-lg-11-92",
                  "grid-lg-12": "MuiGrid-grid-lg-12-93",
                  "grid-lg-2": "MuiGrid-grid-lg-2-83",
                  "grid-lg-3": "MuiGrid-grid-lg-3-84",
                  "grid-lg-4": "MuiGrid-grid-lg-4-85",
                  "grid-lg-5": "MuiGrid-grid-lg-5-86",
                  "grid-lg-6": "MuiGrid-grid-lg-6-87",
                  "grid-lg-7": "MuiGrid-grid-lg-7-88",
                  "grid-lg-8": "MuiGrid-grid-lg-8-89",
                  "grid-lg-9": "MuiGrid-grid-lg-9-90",
                  "grid-lg-auto": "MuiGrid-grid-lg-auto-80",
                  "grid-lg-true": "MuiGrid-grid-lg-true-81",
                  "grid-md-1": "MuiGrid-grid-md-1-68",
                  "grid-md-10": "MuiGrid-grid-md-10-77",
                  "grid-md-11": "MuiGrid-grid-md-11-78",
                  "grid-md-12": "MuiGrid-grid-md-12-79",
                  "grid-md-2": "MuiGrid-grid-md-2-69",
                  "grid-md-3": "MuiGrid-grid-md-3-70",
                  "grid-md-4": "MuiGrid-grid-md-4-71",
                  "grid-md-5": "MuiGrid-grid-md-5-72",
                  "grid-md-6": "MuiGrid-grid-md-6-73",
                  "grid-md-7": "MuiGrid-grid-md-7-74",
                  "grid-md-8": "MuiGrid-grid-md-8-75",
                  "grid-md-9": "MuiGrid-grid-md-9-76",
                  "grid-md-auto": "MuiGrid-grid-md-auto-66",
                  "grid-md-true": "MuiGrid-grid-md-true-67",
                  "grid-sm-1": "MuiGrid-grid-sm-1-54",
                  "grid-sm-10": "MuiGrid-grid-sm-10-63",
                  "grid-sm-11": "MuiGrid-grid-sm-11-64",
                  "grid-sm-12": "MuiGrid-grid-sm-12-65",
                  "grid-sm-2": "MuiGrid-grid-sm-2-55",
                  "grid-sm-3": "MuiGrid-grid-sm-3-56",
                  "grid-sm-4": "MuiGrid-grid-sm-4-57",
                  "grid-sm-5": "MuiGrid-grid-sm-5-58",
                  "grid-sm-6": "MuiGrid-grid-sm-6-59",
                  "grid-sm-7": "MuiGrid-grid-sm-7-60",
                  "grid-sm-8": "MuiGrid-grid-sm-8-61",
                  "grid-sm-9": "MuiGrid-grid-sm-9-62",
                  "grid-sm-auto": "MuiGrid-grid-sm-auto-52",
                  "grid-sm-true": "MuiGrid-grid-sm-true-53",
                  "grid-xl-1": "MuiGrid-grid-xl-1-96",
                  "grid-xl-10": "MuiGrid-grid-xl-10-105",
                  "grid-xl-11": "MuiGrid-grid-xl-11-106",
                  "grid-xl-12": "MuiGrid-grid-xl-12-107",
                  "grid-xl-2": "MuiGrid-grid-xl-2-97",
                  "grid-xl-3": "MuiGrid-grid-xl-3-98",
                  "grid-xl-4": "MuiGrid-grid-xl-4-99",
                  "grid-xl-5": "MuiGrid-grid-xl-5-100",
                  "grid-xl-6": "MuiGrid-grid-xl-6-101",
                  "grid-xl-7": "MuiGrid-grid-xl-7-102",
                  "grid-xl-8": "MuiGrid-grid-xl-8-103",
                  "grid-xl-9": "MuiGrid-grid-xl-9-104",
                  "grid-xl-auto": "MuiGrid-grid-xl-auto-94",
                  "grid-xl-true": "MuiGrid-grid-xl-true-95",
                  "grid-xs-1": "MuiGrid-grid-xs-1-40",
                  "grid-xs-10": "MuiGrid-grid-xs-10-49",
                  "grid-xs-11": "MuiGrid-grid-xs-11-50",
                  "grid-xs-12": "MuiGrid-grid-xs-12-51",
                  "grid-xs-2": "MuiGrid-grid-xs-2-41",
                  "grid-xs-3": "MuiGrid-grid-xs-3-42",
                  "grid-xs-4": "MuiGrid-grid-xs-4-43",
                  "grid-xs-5": "MuiGrid-grid-xs-5-44",
                  "grid-xs-6": "MuiGrid-grid-xs-6-45",
                  "grid-xs-7": "MuiGrid-grid-xs-7-46",
                  "grid-xs-8": "MuiGrid-grid-xs-8-47",
                  "grid-xs-9": "MuiGrid-grid-xs-9-48",
                  "grid-xs-auto": "MuiGrid-grid-xs-auto-38",
                  "grid-xs-true": "MuiGrid-grid-xs-true-39",
                  "item": "MuiGrid-item-12",
                  "justify-xs-center": "MuiGrid-justify-xs-center-28",
                  "justify-xs-flex-end": "MuiGrid-justify-xs-flex-end-29",
                  "justify-xs-space-around": "MuiGrid-justify-xs-space-around-31",
                  "justify-xs-space-between": "MuiGrid-justify-xs-space-between-30",
                  "justify-xs-space-evenly": "MuiGrid-justify-xs-space-evenly-32",
                  "spacing-xs-16": "MuiGrid-spacing-xs-16-34",
                  "spacing-xs-24": "MuiGrid-spacing-xs-24-35",
                  "spacing-xs-32": "MuiGrid-spacing-xs-32-36",
                  "spacing-xs-40": "MuiGrid-spacing-xs-40-37",
                  "spacing-xs-8": "MuiGrid-spacing-xs-8-33",
                  "wrap-xs-nowrap": "MuiGrid-wrap-xs-nowrap-17",
                  "wrap-xs-wrap-reverse": "MuiGrid-wrap-xs-wrap-reverse-18",
                  "zeroMinWidth": "MuiGrid-zeroMinWidth-13",
                }
              }
              component="div"
              container={true}
              direction="row"
              item={false}
              justify="center"
              lg={false}
              md={false}
              sm={false}
              spacing={0}
              wrap="wrap"
              xl={false}
              xs={false}
              zeroMinWidth={false}
            >
              <div
                className="MuiGrid-container-11 MuiGrid-align-items-xs-flex-start-20 MuiGrid-justify-xs-center-28"
              />
            </Grid>
          </WithStyles(Grid)>
        </section>
      </div>
    </Container>
  </WithStyles(Container)>
</Provider>
`;
